activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -1012,17 +1012,18 @@ def invalidate_transaction(exception)
           return unless exception.is_a?(TransactionRollbackError)
           return unless savepoint_errors_invalidate_transactions?
 
-          current_transaction.state.invalidate! if current_transaction
```

Comment:
This change is unrelated but I realized that `current_transaction` is always truthy so this condition is meaningless
https://github.com/rails/rails/blob/999c58d41fe4726ef2b90910cea934f0593aad49/activerecord/lib/active_record/connection_adapters/abstract/transaction.rb#L516

We don't need to perform any other check or make sure that `current_transaction` is not a `NullTransaction` as we can rely on the assumption that we must be in a non-null transaction if we are handling a `TransactionRollbackError` exception

Reply:
> we can rely on the assumption that we must be in a non-null transaction if we are handling a TransactionRollbackError exception

I haven't dug deeper on the surrounding context (so maybe still no action required), but on this particular point, I'm not so sure: `TransactionRollbackException` means "if this occurs inside a transaction, it will get rolled back", not necessarily "this _did_ occur inside a transaction, which has now been rolled back".

Reply:
I'll need to rethink this but here is what I'm planning to do for now:
I'm going to remove this change to this PR to keep only about `retryable_query_error? simplification
And a bit later, in a separate PR I will either remove `if current_transaction` with a better context and investigation or change it to `unless current_transaction.is_a?(NullTransaction)` or any other appropriate check. As we will have to do something in any way as `if current_transaction` does basically nothing and even makes wrong impression